spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/DeploymentPropertiesResolver.java

```diff
@@ -354,15 +355,17 @@ Map<String, String> getNodeSelectors(Map<String, String> deploymentProperties) {
 		Map<String, String> nodeSelectors = new HashMap<>();
 
 		String nodeSelector = this.properties.getNodeSelector();
-
-		String nodeSelectorDeploymentProperty = deploymentProperties
-				.getOrDefault(KubernetesDeployerProperties.KUBERNETES_DEPLOYMENT_NODE_SELECTOR, "");
-
-		boolean hasGlobalNodeSelector = StringUtils.hasText(properties.getNodeSelector());
+		String nodeSelectorDeploymentProperty = deploymentProperties.getOrDefault(KubernetesDeployerProperties.KUBERNETES_DEPLOYMENT_NODE_SELECTOR, "");
+		for (String name : RelaxedNames.forCamelCase(KubernetesDeployerProperties.KUBERNETES_DEPLOYMENT_NODE_SELECTOR)) {
```

Comment:
This solution will not work because `KUBERNETES_DEPLOYMENT_NODE_SELECTOR` does not follow the KubernetesDeployment prefix format of `spring.cloud.deployer.kubernetes`.  It is currently `spring.cloud.deployer.kubernetes.deployment`.  
The addition of the `RelaxedNames` does support when users use camel case instead of kabob case, but with the correct prefix.